diff -ruN byte-buddy-byte-buddy-1.12.6/byte-buddy-dep/src/main/java/net/bytebuddy/dynamic/ClassFileLocator.java byte-buddy-byte-buddy-1.12.6-patched/byte-buddy-dep/src/main/java/net/bytebuddy/dynamic/ClassFileLocator.java
--- byte-buddy-byte-buddy-1.12.6/byte-buddy-dep/src/main/java/net/bytebuddy/dynamic/ClassFileLocator.java	2021-12-20 02:30:58.000000000 -0800
+++ byte-buddy-byte-buddy-1.12.6-patched/byte-buddy-dep/src/main/java/net/bytebuddy/dynamic/ClassFileLocator.java	2023-01-12 09:04:56.000000000 -0800
@@ -476,7 +476,29 @@
              * {@inheritDoc}
              */
             public ClassLoader run() {
-                return new URLClassLoader(new URL[0], ClassLoadingStrategy.BOOTSTRAP_LOADER);
+                return new BootLoaderProxy(new URL[0], ClassLoadingStrategy.BOOTSTRAP_LOADER);
+            }
+
+            /**
+             * A URLClassLoader which exposes addURL to be used as a Bootloader Proxy.
+             */
+            class BootLoaderProxy extends URLClassLoader {
+                /**
+                 * Constructs a new BootLoaderProxy for the given URLs and parent ClassLoader.
+                 * @param      urls the URLs from which to load classes and resources
+                 * @param      parent the parent class loader for delegation
+                 */
+                BootLoaderProxy(URL[] urls, ClassLoader parent) {
+                    super(urls, parent);
+                }
+
+                /**
+                 * {@inheritDoc}
+                 */
+                @Override
+                public void addURL(URL url) {
+                    super.addURL(url);
+                }
             }
         }
 
